cms/static/js/views/manage_users_and_roles.js

```diff
@@ -0,0 +1,306 @@
+                    actions: this.getPossibleRoleChangesForRole(user.role, adminRoleCount),
+                    roles: roles,
+                    allow_delete: !(user.role == this.admin_role.key && adminRoleCount == 1),
+                    allow_actions: this.allow_actions,
+                    is_current_user: is_current_user,
+                    viewHelpers: viewHelpers
+                };
+
+                this.$userList.append(templateFn(template_data));
+            }
+        },
+
+        getAdminRoleCount: function() {
+            var self = this;
+            return _.filter(this.users, function(user) { return user.role == self.admin_role.key; }).length;
+        },
+
+        getPossibleRoleChangesForRole: function(role, adminRoleCount) {
+            var result = [],
+                role_names = _.map(this.roles, function(role) { return role.key });
+            if (role == this.admin_role.key && adminRoleCount==1) {
+                result.push({ notoggle: true });
+            }
+            else {
+                var currentRoleIdx = _.indexOf(role_names, role);
+                // in reverse order to show "Add" buttons to the left, "Remove" to the right
+                for (var i=this.roles.length-1; i>=0; i--) {
+                    var other_role = this.roles[i];
+                    if (Math.abs(currentRoleIdx-i) != 1) continue; // allows moving only to adjacent roles
+                    result.push({
+                        to_role: other_role.key,
+                        label: (i<currentRoleIdx) ? this.roles[currentRoleIdx].name : other_role.name,
+                        direction: (i<currentRoleIdx) ? "remove" : "add"
+                    });
+                }
+            }
+            return result;
+        },
+
+        checkEmail: function(email) {
+            var allUsersEmails = _.map(this.users, function(user) { return user.email; });
+
+            if(!email) {
+                return { valid: false, msg: makeInvalidEmailMessage(this.messages) };
+            }
+
+            if(_.contains(allUsersEmails, email)) {
+                return { valid: false, msg: makeAlreadyMemberMessage(this.messages, email, this.containerName) };
+            }
+            return { valid: true };
+        },
+
+        // Our helper method that calls the RESTful API to add/remove/change user roles:
+        changeRole: function(email, newRole, opts) {
+            var self = this;
+            var url = this.tplUserURL.replace('@@EMAIL@@', email);
+            var errMessage = opts.errMessage || this.messages.defaults.changeRoleError;
+            var onSuccess = opts.onSuccess || function(data){ ViewUtils.reload(); };
+            var onError = opts.onError || function(){};
+            $.ajax({
+                url: url,
+                type: newRole ? 'POST' : 'DELETE',
+                dataType: 'json',
+                contentType: 'application/json',
+                notifyOnError: false,
```

Comment:
What does setting notifyOnError to false here do? Does it prevent the general error handling from taking over?